### ISSUE TITLE
Forms: checked checkbox fails form validation: "on is not one of the available choices"

### ISSUE THREAD

**Issue by oTree-org** [oTree-org/otree-core#427]
In models.py:

```
    volunteer = models.BooleanField(
        widget=widgets.CheckboxInput(),
    )
```

In views.py:

```
class MyPage(Page):
    form_model = models.Player
    form_fields = ['volunteer']
```

Then use `{% formfield player.volunteer %}`.

When I load the page, check the checkbox and submit the form, I get this form validation error:

![image](https://cloud.githubusercontent.com/assets/5423929/12266118/af256334-b984-11e5-85ac-820805ff5a77.png)

**Comment by gregmuellegger** [comment]
The problem here might be that oTree's `BooleanField` inherits from Django's `NullBooleanField`: https://docs.djangoproject.com/en/1.9/ref/models/fields/#django.db.models.NullBooleanField
A `NullBooleanField` has three different states: `True`, `False`, `None`. So a Checkbox is not sufficient in this case.

I think I see the usecase for this here. The BooleanField should have the `None` option since you want to distinguish not yet set fields, but show a checkbox in the frontend so that the user has to decide on yes or no.

I would say that in this case you need a customly implemented CheckboxInput that knows about the oddities of the `NullBooleanField`.

**Comment by oTree-org** [comment]
Agreed. An unchecked checkbox should be False, and a checked checkbox should be True. (Just as it is for Django's BooleanField).

**Comment by gregmuellegger** [comment]
This seems to be an inconsistency in django-floppyforms. It emulated the behaviour of Django in this case, but Django fixed it a few versions back already. We are now using the updated behaviour in django-floppyforms as well. I released django-floppyforms 1.6.1 for this which fixes the problem.
